src/subscription/__tests__/subscribe-test.js

```diff
@@ -0,0 +1,215 @@
+/**
+ *  Copyright (c) 2015, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+import { expect } from 'chai';
+import { describe, it } from 'mocha';
+import EventEmitter from 'events';
+import eventEmitterAsyncIterator from './eventEmitterAsyncIterator';
+import {subscribe} from '../subscribe';
+import { parse } from '../../language';
+import {
+  GraphQLSchema,
+  GraphQLObjectType,
+  GraphQLList,
+  GraphQLBoolean,
+  GraphQLInt,
+  GraphQLString,
+  GraphQLNonNull,
+} from '../../type';
+
+
+describe('Subscribe', () => {
+
+  const EmailType = new GraphQLObjectType({
+    name: 'Email',
+    fields: {
+      from: { type: GraphQLString },
+      subject: { type: GraphQLString },
+      message: { type: GraphQLString },
+      unread: { type: GraphQLBoolean },
+    }
+  });
+
+  const InboxType = new GraphQLObjectType({
+    name: 'Inbox',
+    fields: {
+      total: {
+        type: GraphQLInt,
+        resolve: inbox => inbox.emails.length,
+      },
+      unread: {
+        type: GraphQLInt,
+        resolve: inbox => inbox.emails.filter(email => email.unread).length,
+      },
+      emails: { type: new GraphQLList(EmailType) },
+    }
+  });
+
+  const QueryType = new GraphQLObjectType({
+    name: 'Query',
+    fields: {
+      inbox: { type: InboxType },
+    }
+  });
+
+  const EmailEventType = new GraphQLObjectType({
+    name: 'EmailEvent',
+    fields: {
+      email: { type: EmailType },
+      inbox: { type: InboxType },
+    }
+  });
+
+  const SubscriptionType = new GraphQLObjectType({
+    name: 'Subscription',
+    fields: {
+      importantEmail: { type: EmailEventType },
+    }
+  });
+
+  const emailSchema = new GraphQLSchema({
+    query: QueryType,
+    subscription: SubscriptionType
+  });
+
+  it('produces a payload per subscription event', async () => {
+
+    const pubsub = new EventEmitter();
+
+    const data = {
+      inbox: {
+        emails: [
+          {
+            from: 'joe@graphql.org',
+            subject: 'Hello',
+            message: 'Hello World',
+            unread: false,
+          },
+        ],
+      },
+      importantEmail() {
+        return eventEmitterAsyncIterator(pubsub, 'importantEmail');
```

Comment:
For example, this `data` is the `rootValue`, so when the `importantEmail` field is resolved, it returns an AsyncIterator created from the simple in-memory `EventEmitter` object we're using for this test.